target_decisioning_engine/tests/test_artifact_provider.py

```diff
@@ -0,0 +1,257 @@
+# Copyright 2020 Adobe. All rights reserved.
+# This file is licensed to you under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License. You may obtain a copy
+# of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+# OF ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+"""Unit tests for target_decisioning_engine.artifact_provider module"""
+# pylint: disable=too-many-public-methods
+import json
+import unittest
+import time
+import urllib3
+
+try:
+    from unittest.mock import patch, Mock
+except ImportError:
+    from mock import patch, Mock
+try:
+    from http.client import NOT_MODIFIED, OK
+except ImportError:
+    from httplib import NOT_MODIFIED, OK
+
+from urllib3.exceptions import MaxRetryError
+from urllib3.response import HTTPResponse
+from target_decisioning_engine.artifact_provider import ArtifactProvider
+from target_decisioning_engine.constants import MINIMUM_POLLING_INTERVAL
+from target_decisioning_engine.constants import DEFAULT_POLLING_INTERVAL
+from target_decisioning_engine.types.decisioning_config import DecisioningConfig
+from target_decisioning_engine.events import ARTIFACT_DOWNLOAD_FAILED
+from target_decisioning_engine.events import ARTIFACT_DOWNLOAD_SUCCEEDED
+from target_python_sdk.tests.helpers import spy_decorator
+
+ARTIFACT_URL = "http://my.artifact.com"
+
+
+class TestArtifactProvider(unittest.TestCase):
+
+    def setUp(self):
+        self.default_config = DecisioningConfig("client123", "org999", artifact_location=ARTIFACT_URL,
+                                                event_emitter=Mock())
+        self.provider = ArtifactProvider(self.default_config)
```

Comment:
I see `self.provider` being set to a new provider in the tests below.  So would this provide continue to poll or it would be garbage collected?  Maybe it's not an issue, because the program exits already.